lib/extensions/attachable/active_record/base.rb

```diff
@@ -2,31 +2,102 @@
 module Extensions::Attachable::ActiveRecord::Base
   module ClassMethods
     # This function should be declared in model, to it have attachments.
-    def has_many_attachments # rubocop:disable Style/PredicateName
+    #
+    # @param [Hash] options
+    # @option options [Symbol|Array<Symbol>] :on The column which is associated with attachments,
+    #   the column type should be string or text.
+    #   This can be a symbol or an array of symbols.
+    # @example Has many attachments on a column
+    #   has_many_attachments on: :description #=> description is associated with the attachments
+    #   of the model, updating description will result in attachments changing.
+    def has_many_attachments(options = {}) # rubocop:disable Style/PredicateName
+      include HasManyAttachments
+
+      class_attribute :attachable_columns
+      if options[:on]
+        self.attachable_columns = Array(options[:on])
```

Comment:
Yes, I am going to add that in a later PR

On Saturday, 30 April 2016, Weiqing <notifications@github.com> wrote:

> In lib/extensions/attachable/active_record/base.rb
> <https://github.com/Coursemology/coursemology2/pull/1039#discussion_r61621867>
> :
>
> > @@ -2,31 +2,102 @@
> >  module Extensions::Attachable::ActiveRecord::Base
> >    module ClassMethods
> >      # This function should be declared in model, to it have attachments.
> > -    def has_many_attachments # rubocop:disable Style/PredicateName
> > +    #
> > +    # @param [Hash] options
> > +    # @option options [Symbol|Array<Symbol>] :on The column which is associated with attachments,
> > +    #   the column type should be string or text.
> > +    #   This can be a symbol or an array of symbols.
> > +    # @example Has many attachments on a column
> > +    #   has_many_attachments on: :description #=> description is associated with the attachments
> > +    #   of the model, updating description will result in attachments changing.
> > +    def has_many_attachments(options = {}) # rubocop:disable Style/PredicateName
> > +      include HasManyAttachments
> > +
> > +      class_attribute :attachable_columns
> > +      if options[:on]
> > +        self.attachable_columns = Array(options[:on])
>
> Do we need to check if the columns are valid?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/1039/files/c9355a4b35858d51573c5331507969b4318ced52#r61621867>
>